cypher-shell/src/main/java/org/neo4j/shell/cli/InteractiveShellRunner.java

```diff
@@ -145,10 +157,24 @@ AnsiFormattedText getPrompt() {
         if (statementParser.containsText()) {
             return continuationPrompt;
         }
-        if (txHandler.isTransactionOpen()) {
-            return transactionPrompt;
-        }
-        return freshPrompt;
+
+        String databaseName = databaseManager.getActiveDatabase();
+
+        // Substitute empty name for the default default-database-name
+        // For now we just use a hard-coded default name
+        // Ideally we would like to receive the actual name in the ResultSummary when we connect (in BoltStateHandler.reconnect())
+        // (If the user is an admin we could also query for the default database config value with:
+        //   "CALL dbms.listConfig() YIELD name, value WHERE name = "dbms.default_database" RETURN value"
+        //  but that does not work in general)
+        databaseName = ABSENT_DB_NAME.equals(databaseName) ? DEFAULT_DEFAULT_DB_NAME : databaseName;
+
+        AnsiFormattedText prompt = AnsiFormattedText.s().bold()
+                .append(connectionConfig.username())
+                .append("@")
+                .append(databaseName)
+                .appendNewLine()
```

Comment:
This is a change to the formatting. Will everyone want this newline?

Reply:
Let me explain the reason why I introduced this:

Before the prompt always had a fixed (and _short_ length), and for multi-line statements we would use our knowledge of this fixed length to indent subsequent lines in a multi-line statement so that all clauses were nicely aligned. 

Now we have a prompt length that can vary. We could compute the current indent amount and keep the same alignment, but that got me thinking about the case where username + database name is very long: do you really want to lose all that console line estate? This is also the main reason I prefer to use a newline in an ordinary informative shell-prompt (that shows working directory path etc.); so you don't have to type your commands at the far-edge of the window.

Alternatively, we could skip the indentation of subsequent lines, but that also makes queries look a bit ugly (I played around with both).

This was the quickest way to solve two problems (long prompts and multi-line statement aligned indentation). It is a fair point that this could be configurable, but I still think this makes for the preferred default.

Reply:
OK. I personally prefer the single line with calculated indent, but my preference is not especially strong, so I won't argue :-)